frontend/packages/dev-console/src/utils/__tests__/resource-label-utils.spec.tsx

```diff
@@ -0,0 +1,16 @@
+import { mergeData } from '../resource-label-utils';
+import { newDeployment, originalDeployment, deployment } from './resource-label-utils-data';
+
+describe('resource-label-utils', () => {
+  describe('mergeData', () => {
+    it('should return newResource when originalResource is empty', () => {
+      const mergedResource = mergeData({}, newDeployment);
+      expect(mergedResource).toEqual(newDeployment);
+    });
+
+    it('should return the mergedResource with correct data', () => {
+      const mergedResource = mergeData(originalDeployment, newDeployment);
```

Comment:
This is good initial tests, but we should probably have a test (`it`) for each if block.

Reply:
I believe we should still get more tests here. If nothing more than a specific test for volumes and a test for no volumes (ie the fix you did).

Reply:
Sure, will add more tests.